api/src/wfl/service/postgres.clj

```diff
@@ -55,21 +56,29 @@
   "Test if a workflow `:status` is in a terminal state."
   (set (conj cromwell/final-statuses "skipped")))
 
-(defn update-workflow-statuses!
+(defn ^:private make-update-workflows [get-status!]
   "Use `tx` to update `status` of `workflows` in `_workload`."
-  [tx {:keys [cromwell items workflows] :as _workload}]
-  (letfn [(maybe-cromwell-status [{:keys [uuid]}]
+  (fn [tx {:keys [items workflows] :as workload}]
+    (letfn [(update! [{:keys [id uuid]} status]
+              (jdbc/update! tx items
+                            {:updated (OffsetDateTime/now) :uuid uuid :status status}
+                            ["id = ?" id]))]
+      (->> workflows
+           (remove (comp nil? :uuid))
+           (remove (comp finished? :status))
+           (run! #(update! % (get-status! workload %)))))))
+
+(def update-workflow-statuses!
```

Comment:
Can you move the doc string from the private function  (`make-update-workflows`) onto these two new implementations? Don't think the private function has much business having a doc string....

Reply:
Good point!